src/ObjWriting/XModel/Export/XModelBinWriter.cpp

```diff
@@ -0,0 +1,484 @@
+            XModelBinWriterBase::Write(material.ambientColor[3]); // A
+
+            XModelBinWriterBase::Write(XModelBinHash::MATERIAL_INCANDESCENCE);
+            XModelBinWriterBase::Write(material.incandescence[0]);
+            XModelBinWriterBase::Write(material.incandescence[1]);
+            XModelBinWriterBase::Write(material.incandescence[2]);
+            XModelBinWriterBase::Write(material.incandescence[3]);
+
+            XModelBinWriterBase::Write(XModelBinHash::MATERIAL_COEFFS);
+            XModelBinWriterBase::Write(material.coeffs[0]);
+            XModelBinWriterBase::Write(material.coeffs[1]);
+
+            XModelBinWriterBase::Write(XModelBinHash::MATERIAL_GLOW);
+            XModelBinWriterBase::Write(material.glow.x);
+            XModelBinWriterBase::Write(material.glow.y);
+
+            XModelBinWriterBase::Write(XModelBinHash::MATERIAL_REFRACTIVE);
+            XModelBinWriterBase::Write(material.refractive.x);
+            XModelBinWriterBase::Write(material.refractive.y);
+
+            XModelBinWriterBase::Write(XModelBinHash::MATERIAL_SPECULAR_COLOR);
+            XModelBinWriterBase::Write(material.specularColor[0]); // R
+            XModelBinWriterBase::Write(material.specularColor[1]); // G
+            XModelBinWriterBase::Write(material.specularColor[2]); // B
+            XModelBinWriterBase::Write(material.specularColor[3]); // A
+
+            XModelBinWriterBase::Write(XModelBinHash::MATERIAL_REFLECTIVE_COLOR);
+            XModelBinWriterBase::Write(material.reflectiveColor[0]); // R
+            XModelBinWriterBase::Write(material.reflectiveColor[1]); // G
+            XModelBinWriterBase::Write(material.reflectiveColor[2]); // B
+            XModelBinWriterBase::Write(material.reflectiveColor[3]); // A
+
+            XModelBinWriterBase::Write(XModelBinHash::MATERIAL_REFLECTIVE);
+            XModelBinWriterBase::Write(material.reflective.x);
+            XModelBinWriterBase::Write(material.reflective.y);
+
+            XModelBinWriterBase::Write(XModelBinHash::MATERIAL_BLINN);
+            XModelBinWriterBase::Write(material.blinn[0]);
+            XModelBinWriterBase::Write(material.blinn[1]);
+
+            XModelBinWriterBase::Write(XModelBinHash::MATERIAL_PHONG);
+            XModelBinWriterBase::Write(material.phong);
+
+            materialNum++;
+        }
+    }
+
+public:
+    XModelBinWriter7(std::ostream& stream, std::string gameName, std::string zoneName)
+        : XModelBinWriterBase(stream, std::move(gameName), std::move(zoneName))
+    {
+    }
+
+    void Write(const XModelCommon& xmodel) override
+    {
+        PrepareVertexMerger(xmodel);
+        WriteHeader(7);
+        WriteBones(xmodel);
+        WriteVertices(xmodel);
+        WriteFaces(xmodel);
+        WriteObjects(xmodel);
+        WriteMaterials(xmodel);
+
+        auto estimatedCompressedFileSize = LZ4_compressBound(m_writer.str().size());
+        auto compressedBuffer = new char[estimatedCompressedFileSize];
```

Comment:
You are leaking memory here, you are doing `new`, but not `delete`.

Best do
```cpp
const auto compressedBuffer = std::make_unique<char[]>(estimatedCompressedFileSize);
```
here, then the allocated buffer is automatically freed when the compressedBuffer variable leaves scope.